### ISSUE TITLE
Listbox accessibility problem

### ISSUE THREAD

**Issue by afr-ux** [home-assistant/frontend#24618]
### Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.
- [x] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Accessibility problem with list box fields.

### Describe the behavior you expected

Screenreader say every item in the list

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

All

### In which browser are you experiencing the issue?

Chrome, firefox, Edge

### Which operating system are you using to run this browser?

Windows 10 Pro

### State of relevant entities

```txt

```

### Problem-relevant frontend configuration

```yaml

```

### Javascript errors shown in your browser console/inspector

```txt

```

### Additional information

The ‘list box’ component is not accessible with screen readers.
This component is in several places. For example, when I click on the ‘create helper’ button, there is a list box to select a type from all the possible ones. This component is not accessible with screen readers because it is not well constructed.
You can make a list component accessible or, for example, a standard combobox, since the objective is to be able to select one element among several and a combobox has that function.
Thanks.

**Comment by karwosts** [comment]
I believe this create helper dialog became non-functional in 2025.3. The linked PR I hope should restore functionality of the create helper dialog back to how it was in previous versions. 

You alluded to similar issues that might exist in other places, if you find more please let us know where you find similar problems.
